addons/mail/static/src/chatter/web/form_compiler.js

```diff
@@ -73,15 +72,16 @@ patch(FormCompiler.prototype, {
         }
 
         const webClientViewAttachmentViewHookXml = res.querySelector(".o_attachment_preview");
+        const hasPreview = !!webClientViewAttachmentViewHookXml;
```

Comment:
`hasAttachmentPreview` could be stored so that we don't require to pass it as parameter to `mailLayout()`, which would make the conditional easier to read

Reply:
Stored where?

Reply:
Yeah whatever form compiler/renderer sucks so let's keep it like this.

In theory form renderer has access to this compiled templates `res`, in `FormRenderer.templates["FormRenderer"]`. So I think `this.templates.FormRenderer.querySelector(".o_attachment_preview")"` would work... But this is overly complicated :(